util/rate_limiter_test.cc

```diff
@@ -117,12 +119,26 @@ TEST_F(RateLimiterTest, Rate) {
               arg.request_size - 1, target / 1024, rate / 1024,
               elapsed / 1000000.0);
 
-      ASSERT_GE(rate / target, 0.80);
+      ++samples;
+      if (rate / target >= 0.80) {
+        ++samples_at_minimum;
+      }
       ASSERT_LE(rate / target, 1.25);
     }
   }
-}
+
+  // This can fail in heavily loaded CI environments
+  if (getenv("SANDCASTLE")
+#if (defined(TRAVIS) || defined(CIRCLECI)) && defined(OS_MACOSX)
+      || true
 #endif
+  ) {
+    fprintf(stderr, "Skipped minimum rate check (%d / %d passed)\n",
+            samples_at_minimum, samples);
```

Comment:
Should there also be a ROCKSDB_GTEST_SKIP here, so we can somehow potentially in the future know that the test was skipped?

Reply:
We're only skipping one aspect of the test, so I don't think ROCKSDB_GTEST_SKIP is appropriate here. I printed out the line above for more visibility into something being skipped.

Let me know if I have done anything to suggest that this is a good test, or to make it more difficult for the next person to work on RateLimiter tests to improve things further.